### PR TITLE
Fix output tensor shape for argmin and argmax where keepdim=True and dim=None

### PR DESCRIPTION
**Pull request by mrnikwaws**
Current failure looks like this:

```
root@f825750bf417:/ansible/torch_xla/pytorch/xla# export PJRT_DEVICE="CPU"
root@f825750bf417:/ansible/torch_xla/pytorch/xla# python
Python 3.8.18 (default, Feb  1 2024, 06:10:58) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> import torch_xla
>>> t = torch.rand((3,4))
>>> t
tensor([[0.9253, 0.5503, 0.4175, 0.7273],
        [0.8306, 0.7907, 0.2054, 0.5639],
        [0.4089, 0.1673, 0.9702, 0.4839]])
>>> t_xla = t.to('xla')
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1707952093.212406   88634 cpu_client.cc:404] TfrtCpuClient created.
>>> t_xla
tensor([[0.9253, 0.5503, 0.4175, 0.7273],
        [0.8306, 0.7907, 0.2054, 0.5639],
        [0.4089, 0.1673, 0.9702, 0.4839]], device='xla:0')
>>> torch.argmax(t,dim=None,keepdim=True)
tensor([[10]])
>>> torch.argmax(t_xla,dim=None,keepdim=True)
tensor(10, device='xla:0'
```

After:
```
root@f825750bf417:/ansible/torch_xla/pytorch/xla# python
Python 3.8.18 (default, Feb  1 2024, 06:10:58) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> import torch_xla
>>> t = torch.randn((3,4))
>>> t
tensor([[-1.6839e+00, -5.5569e-01, -1.1452e+00,  4.5730e-01],
        [ 7.5517e-01,  2.3971e+00,  5.8805e-01,  8.4879e-01],
        [-3.3246e-04,  1.4524e-01,  2.0454e-01, -5.7229e-01]])
>>> t_xla = t.to('xla')
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1707953861.897586   92883 cpu_client.cc:404] TfrtCpuClient created.
>>> torch.argmax(t,keepdim=True)
tensor([[5]])
>>> torch.argmax(t_xla,keepdim=True)
tensor([[5]], device='xla:0')
>>> torch.argmin(t,keepdim=True)
tensor([[0]])
>>> torch.argmin(t_xla,keepdim=True)
tensor([[0]], device='xla:0')
```

### CODE REVIEW

**Review by wonjoolee95 — CHANGES_REQUESTED** [review]
Thanks! Changes LGTM, can you add a corresponding unit test at https://github.com/pytorch/xla/blob/master/test/cpp/test_aten_xla_tensor_2.cpp#L2077 and fix the linter issues?

**Comment by mrnikwaws** [comment]
Yep will do

**Review by wonjoolee95 — APPROVED** [review]
Thanks! Changes LGTM, let's wait for the CI to verify the tests.

**Comment by mrnikwaws** [comment]
I misread the test code - they need unique names - fixing now

**Comment by wonjoolee95** [comment]
Apologies for the delay but can you rebase with this head? The failing tests have been disabled on head, but I hope to let the CI complete before merging this. Thanks!

**Comment by mrnikwaws** [comment]
Looks like new unrelated failures

**Comment by wonjoolee95** [comment]
Yeah, it's been a rough day for the head CI.. 😢 

Let's just wait for the other CIs and merge it if they're green, changes LGTM anyways. Thanks!

**Comment by wonjoolee95** [comment]
CI seems to be passing other than the same failure on head. Merging this. Thanks!
